### PR TITLE
feat: add convenient selection of data parallelism

### PR DESCRIPTION
**Pull request by shpface**
*Issue #, if available:*

*Description of changes:*
Adds a parameter to easily populate the hyperparameters of CreateJob for the use of sagemaker data parallelism.

*Testing done:*
unit tests, integ tests

## Merge Checklist

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your pull request._

#### General

- [x] I have read the [CONTRIBUTING](https://github.com/aws/amazon-braket-sdk-python/blob/main/CONTRIBUTING.md) doc
- [x] I used the commit message format described in [CONTRIBUTING](https://github.com/aws/amazon-braket-sdk-python/blob/main/CONTRIBUTING.md#commit-your-change)
- [x] I have updated any necessary documentation, including [READMEs](https://github.com/aws/amazon-braket-sdk-python/blob/main/README.md) and [API docs](https://github.com/aws/amazon-braket-sdk-python/blob/main/CONTRIBUTING.md#documentation-guidelines) (if appropriate)

#### Tests

- [x] I have added tests that prove my fix is effective or that my feature works (if appropriate)
- [x] I have checked that my tests are not configured for a specific region or account (if appropriate)

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: src/braket/jobs/quantum_job_creation.py [hunk 1]
```diff
@@ -191,6 +196,12 @@ def prepare_quantum_job(
             "s3Uri"
         ]
         aws_session.copy_s3_directory(checkpoints_to_copy, checkpoint_config.s3Uri)
+    if distribution == "dataparallel":
+        distribruted_hyperparams = {
+            "sagemaker_distributed_dataparallel_enabled": "true",
+            "sagemaker_instance_type": instance_config.instanceType,
+        }
+        hyperparameters.update(distribruted_hyperparams)
```
Comment (ajberdy) [inline src/braket/jobs/quantum_job_creation.py:204]:
```suggestion
        distributed_hyperparams = {
            "sagemaker_distributed_dataparallel_enabled": "true",
            "sagemaker_instance_type": instance_config.instanceType,
        }
        hyperparameters.update(distributed_hyperparams)
```

File: src/braket/aws/aws_quantum_job.py [hunk 2]
```diff
@@ -134,6 +135,10 @@ def create(
                 to execute the job. Default: InstanceConfig(instanceType='ml.m5.large',
                 instanceCount=1, volumeSizeInGB=30).
 
+            distribution (str): A str that specifies how the job should be distributed. If set to
```
Comment (ajberdy) [inline src/braket/aws/aws_quantum_job.py:138]: Should we add a note that it's intended for use with >1 instance count?

Comment (shpface) [reply src/braket/aws/aws_quantum_job.py:138]: I don't think so. Data parallel distribution could also be used with a single multi-gpu instance

Comment (ajberdy) [reply src/braket/aws/aws_quantum_job.py:138]: Ah, I didn't realize that. In that case, is there any potential a user may want to create a local job with data parallel distribution if their local hardware supports it?

Comment (yitchen-tim) [reply src/braket/aws/aws_quantum_job.py:138]: I think we can cut local mode out of scope. According to this [link](https://sagemaker.readthedocs.io/en/stable/overview.html), SageMaker local mode does not support distributed training with local GPU

**Review by ajberdy — APPROVED** [review]

**Review by yitchen-tim — APPROVED** [review]

**Review by christianbmadsen — COMMENTED** [review]
Let's change the name from distribution to data_parallel

**Comment by shpface** [comment]
> Let's change the name from distribution to data_parallel

The option is kept as `distribution : str` rather than `data_parallel : bool` to leave the door open to future distribution methods.

**Pull request merged**